### PR TITLE
[Wait for #1733] [nn] Attach activation realizer

### PR DESCRIPTION
**Pull request by zhoonit**
## Dependency of the PR

- **[Pending commit: #1733]**

## Commits to be reviewed in this PR

<details><summary>[Realizer] Implement activation realizer</summary><br />

This patch implement activation realizer

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

</details>

<details><summary>[Test] Activation realizer test</summary><br />

This patch implement activation realizer

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

</details>

<details><summary>[nn] Attach activation realizer</summary><br />

This patch attach activation realizer

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

</details>

<details><summary>[Clean] Remove unused function</summary><br />

remove unused function before adaption multioutput.

Main reason for the clean is that the functions are not used but
hard to maintain for the incoming change of adapting multi output

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

</details>

### CODE REVIEW

File: nntrainer/models/neuralnet.h [hunk 1]
```diff
@@ -318,16 +318,6 @@ class NeuralNetwork : public ml::train::Model {
...
 
-  /**
-   * @brief     join passed graph into the existing graph model
-   * @param[in] graph graph to be added/to extend
-   * @param[in] prefix prefix added to names of layers from this graph
-   * @note It is assumed that this model is valid by itself
-   * @retval #ML_ERROR_NONE Successful.
-   * @retval #ML_ERROR_INVALID_PARAMETER invalid parameter.
-   */
-  int extendGraph(GraphType graph, std::string prefix = "");
```
Comment (jijoongmoon) [inline nntrainer/models/neuralnet.h:329]: I thought it is used to extend or merge backbone network into the current network.

Comment (zhoonit) [reply nntrainer/models/neuralnet.h:329]: `addWithReferenceLayers` now takes the job so removed to avoid unmaintained code being present.

**Review by kparichay — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**